### ISSUE TITLE
Rewire API added to module twice

### ISSUE THREAD

**Issue by mbfisher** [speedskater/babel-plugin-rewire#57]
A contrived example:

```js
MyClass.js

export default class MyClass {
}
```

```js
index.js

const MyClass = require('./MyClass');
MyClass.foo = 'bar';

export default MyClass;
```

This leads to an error whereby `__Rewire__` is added to the `MyClass` export in `MyClass.js` **and** `index.js`, causing `Object.defineProperty` to fail the second time around:

```js
Object.defineProperty(_defaultExport, '__Rewire__', {
    'value': __Rewire__,
    'enumberable': false
});
```

I'd love have a go at a PR for this but have never worked with the Babel internals and it looks a bit mad...

M

**Comment by TheSavior** [comment]
Interesting. I'd imagine this could be a problem with any of the rewire flavors. I recently ran into this error with rewire-global but didn't look into it yet. @i-like-robots, have you come across this? @jhnns, perhaps we should be adding a check that these functions don't already exist in https://github.com/jhnns/rewire/blob/master/lib/getDefinePropertySrc.js

This is further reason why we should figure out how to be sharing all this code between all these modules ala https://github.com/i-like-robots/rewireify/issues/17

**Comment by mbfisher** [comment]
I have [a commit](https://github.com/speedskater/babel-plugin-rewire/compare/master...mbfisher:issue-57) which fixes my instance of this. I've been able to test the changes in `exit` but not in `ExportDefaultDeclaration`. I've never worked with the Babel API before - is it along the right lines?

M

**Comment by speedskater** [comment]
@mbfisher thanks for reporting this issue and creating the PR. Your issue is resolved by https://github.com/speedskater/babel-plugin-rewire/pull/62. 

Generally patterns like in your sample were an existing class is changed and then reexported should be avoided, as this always modifies the original class. If this is intended and you want to be able to rewire the modified as well as the original module it is recommended to use the __RewireApi__ object.

**Comment by mbfisher** [comment]
Cheers!

M

On Thursday, 1 October 2015, speedskater <notifications@github.com> wrote:

> @mbfisher <https://github.com/mbfisher> thanks for reporting this issue
> and creating the PR. Your issue is resolved by #62
> <https://github.com/speedskater/babel-plugin-rewire/pull/62>.
>
> Generally patterns like in your sample were an existing class is changed
> and then reexported should be avoided, as this always modifies the original
> class. If this is intended and you want to be able to rewire the modified
> as well as the original module it is recommended to use the *RewireApi*
> object.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/speedskater/babel-plugin-rewire/issues/57#issuecomment-144770682>
> .
>


-- 
Mike Fisher

Email: mike@mbfisher.com
Skype: mb_fisher
GitHub: mbfisher
